service/src/main/java/bio/terra/axonserver/utils/CloudStorageUtils.java

```diff
@@ -77,6 +85,15 @@ public static byte[] getBucketObject(
             .getService()
             .reader(blobId)) {
 
+      Set<PosixFilePermission> permissions = PosixFilePermissions.fromString("rwx------");
+      File outputFile =
+          File.createTempFile(
+              FilenameUtils.getBaseName(objectName) + "-",
+              "." + FilenameUtils.getExtension(objectName),
+              new File(CreateDownloadsFolder.DOWNLOADS_DIR.toString()));
+      Files.setPosixFilePermissions(Paths.get(outputFile.getPath()), permissions);
+      FileOutputStream outputStream = new FileOutputStream(outputFile);
```

Comment:
Suggest putting this in a resource `for` so it will auto-close, instead of relying on the explicit call.
Also, is it possible to set `deleteOnExit` either as part of the output file create or immediately following? Having it at the end means a throw out of this method will not have it set, right?

Reply:
Done. You're right.. now calling `deleteOnExit` immediately after creating.